File: src/main/java/dev/zprestige/fire/module/movement/antivoid/MoveListener.java
package dev.zprestige.fire.module.movement.antivoid;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.MoveEvent;

public class MoveListener extends EventListener<MoveEvent, AntiVoid> {

    public MoveListener(final AntiVoid antiVoid){
        super(MoveEvent.class, antiVoid);
    }

    @Override
    public void invoke(final Object object){
        final MoveEvent event = (MoveEvent) object;
        if (!module.alreadyInVoid && mc.player.posY <= 0.0f) {
            event.setCancelled();
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/antivoid/MoveListener.kt
package dev.zprestige.fire.module.movement.antivoid

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.MoveEvent


class MoveListener(antiVoid: AntiVoid) : EventListener<MoveEvent, AntiVoid>(MoveEvent::class.java, antiVoid) {
    
    override fun invoke(e: Any) {
        val event = e as MoveEvent
        if (!module.alreadyInVoid && mc.player.posY <= 0.0f) {
            event.setCancelled()
        }
    }
}